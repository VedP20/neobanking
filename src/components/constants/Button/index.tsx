import {
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';

interface TButton {
  title: string;
  iconName?: string;
  iconColor?: string;
  variant?: 'Filled' | 'Outlined' | 'icon';
  buttonStyle?: ViewStyle;
  textStyle?: TextStyle;
}

const Button = ({
  title,
  iconName,
  iconColor,
  variant,
  textStyle,
  buttonStyle,
  ...props
}: TButton) => {
  const buttonStyles = [
    styles.container,
    variant === 'Filled' && styles.filledButton,
    variant === 'Outlined' && styles.outlinedButton,
    variant === 'icon' && [styles.iconButton, {paddingHorizontal: 0}],
    buttonStyle,
  ];

  const textStyles = [
    styles.titleText,
    variant === 'Filled' && styles.filledText,
    variant === 'Outlined' && styles.outlinedText,
    textStyle,
  ];

  return (
    <TouchableOpacity style={buttonStyles} {...props}>
      <Text style={textStyles}>{title}</Text>
      {iconName && <Icon name={iconName} size={16} color={iconColor} />}
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'black',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 6,
    height: 46,
    width: '100%',
    justifyContent: 'center',
    borderRadius: 48,
  },
  titleText: {
    color: 'white',
    fontSize: 16,
    lineHeight: 16,
  },
  filledButton: {
    backgroundColor: 'black',
  },
  outlinedButton: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: 'black',
    borderRadius: 12,
  },
  iconButton: {},
  filledText: {},
  outlinedText: {
    color: 'black',
  },
});
