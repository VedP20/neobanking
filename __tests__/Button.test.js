/**
 * @format
 */

import 'react-native';
import React from 'react';
import Button from '../src/components/constants/Button/index';

// Note: import explicitly to use the types shiped with jest.
import { it } from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', async () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
