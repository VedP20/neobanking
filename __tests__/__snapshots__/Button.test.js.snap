// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 48,
      "flexDirection": "row",
      "gap": 6,
      "height": 46,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 16,
          "lineHeight": 16,
        },
        false,
        false,
        undefined,
      ]
    }
  />
</View>
`;
